### PR TITLE
Add Base64Url fuzzer

### PR DESCRIPTION
**Pull request by buyaa-n**
Add Base64UrlFuzzer. 

Mostly covered char overloads as byte overloads quite similar and can be covered with Base64Fuzzer

### CODE REVIEW

File: src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/Base64UrlFuzzer.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Text;
+using System.Formats.Asn1;
+using System.Reflection;
+
+namespace DotnetFuzzing.Fuzzers
+{
+    internal class Base64UrlFuzzer : IFuzzer
```
Comment (WeihanLi) [inline src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/Base64UrlFuzzer.cs:11]: Maybe better to be `sealed`?
```suggestion
    internal sealed class Base64UrlFuzzer : IFuzzer
```

File: src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/Base64UrlFuzzer.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        public void FuzzTarget(ReadOnlySpan<byte> bytes)
+        {
+            using PooledBoundedMemory<byte> inputPoisoned = PooledBoundedMemory<byte>.Rent(bytes, PoisonPagePlacement.After);
+            Span<byte> input = inputPoisoned.Span;
+            int encodedLength = Base64Url.GetEncodedLength(bytes.Length);
+            using PooledBoundedMemory<char> destPoisoned = PooledBoundedMemory<char>.Rent(encodedLength, PoisonPagePlacement.After);
+            Span<char> encoderDest = destPoisoned.Span;
+            int maxDecodedLength = Base64Url.GetMaxDecodedLength(encodedLength);
+            using PooledBoundedMemory<byte> decoderDestPoisoned = PooledBoundedMemory<byte>.Rent(maxDecodedLength, PoisonPagePlacement.After);
+            Span<byte> decoderDest = decoderDestPoisoned.Span;
```
Comment (stephentoub) [inline src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/Base64UrlFuzzer.cs:26]: Nit: this is pretty dense and hard to read... maybe:
```suggestion
            using PooledBoundedMemory<byte> inputPoisoned = PooledBoundedMemory<byte>.Rent(bytes, PoisonPagePlacement.After);
            int encodedLength = Base64Url.GetEncodedLength(bytes.Length);
            int maxDecodedLength = Base64Url.GetMaxDecodedLength(encodedLength);

            using PooledBoundedMemory<char> destPoisoned = PooledBoundedMemory<char>.Rent(encodedLength, PoisonPagePlacement.After);
            using PooledBoundedMemory<byte> decoderDestPoisoned = PooledBoundedMemory<byte>.Rent(maxDecodedLength, PoisonPagePlacement.After);

            Span<byte> input = inputPoisoned.Span;
            Span<char> encoderDest = destPoisoned.Span;
            Span<byte> decoderDest = decoderDestPoisoned.Span;

```

**Review by stephentoub — APPROVED** [review]

**Comment by buyaa-n** [comment]
@MihuBot fuzz base64url

**Comment by buyaa-n** [comment]
The input that failed the fuzzer is `Base64Url.IsValid("SM=")` which supposed to return false, but returning true, the fix is up https://github.com/dotnet/runtime/pull/106771

**Comment by buyaa-n** [comment]
Last [fuzz run](https://github.com/MihuBot/runtime-utils/issues/563) with the bug fix https://github.com/dotnet/runtime/pull/106771 succeeded, merging

**Pull request merged**